### PR TITLE
GPU Opensource Observability Pattern

### PR DESCRIPTION
**Pull request by freschri**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
+# Single New EKS Cluster Open Source Observability Accelerator for NVIDIA GPU-based clusters
+
+Graphics Processing Units (GPUs) play an integral part in the Machine Learning (ML) workflow, by providing the scalable performance needed for fast ML training and cost-effective ML inference. On top of that, they are used in flexible remote virtual workstations and powerful HPC computations.
+
+This pattern shows you how to monitor the performance of the GPUs units, used in an Amazon EKS cluster leveraging GPU-based instances.
+
+Prometheus and Grafana are open source tools used to collect and visualise metrics respectively.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:7]: Lets talk about AMG and AMP here.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+The AMI type of the node group is `AL2_x86_64_GPU AMI`, which uses the [Amazon EKS-optimized Linux AMI with GPU support](https://aws.amazon.com/marketplace/pp/prodview-nwwwodawoxndm). In addition to the standard Amazon EKS-optimized AMI configuration, the GPU AMI includes the NVIDIA drivers.
+
+The [NVIDIA Data Center GPU Manager](https://docs.nvidia.com/data-center-gpu-manager-dcgm/index.html) (DCGM) is a suite of tools for managing and monitoring NVIDIA datacenter GPUs in cluster environments. It includes health monitoring, diagnostics, system alerts and governance policies.
+GPU metrics are exposed to Prometheus by the [DCGM Exporter](https://github.com/NVIDIA/dcgm-exporter), that uses the Go bindings to collect GPU telemetry data from DCGM and then exposes the metrics for Prometheus to pull from, using an http endpoint (`/metrics`).
+
+The pattern deploys the [NVIDIA GPU Operator add-on](https://aws-quickstart.github.io/cdk-eks-blueprints/addons/gpu-operator/). The [GPU Operator](https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/overview.html) uses the NVIDIA DCGM Exporter to expose GPU telemetry to Prometheus.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:18]: lets call this AMP instead of Prometheus

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+
+Data is visualised in Grafana by the [NVIDIA DCGM Exporter Dashboard](https://grafana.com/grafana/dashboards/12239-nvidia-dcgm-exporter-dashboard).
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:20]: Same call it AMG

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+    
+## Teardown
+
+You can teardown the whole CDK stack with the following command:
+
+```bash
+make pattern single-new-eks-gpu-opensource-observability destroy
+```
+
+## Troubleshooting
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:258]: This section is repeatative and can be removed.

File: lib/single-new-eks-opensource-observability-pattern/gpu-index.ts [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+        ObservabilityBuilder.builder()
+            .account(account)
+            .region(region)
+            .version('auto')
+            .enableOpenSourcePatternAddOns(ampAddOnProps)
+            .resourceProvider(ampWorkspaceName, new blueprints.CreateAmpProvider(ampWorkspaceName, ampWorkspaceName))
+            .clusterProvider(
+                new blueprints.GenericClusterProvider({
+                    version: eks.KubernetesVersion.V1_27,
```
Comment (elamaran11) [inline lib/single-new-eks-opensource-observability-pattern/gpu-index.ts:82]: Why is this required when. you are providing `version('auto')`

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@freschri Great Work some minor doc feedback. Please check on it.

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-gpu-opensource-observability deploy

File: lib/single-new-eks-opensource-observability-pattern/gpu-index.ts (hunk 5 above)
Comment (freschri) [reply lib/single-new-eks-opensource-observability-pattern/gpu-index.ts:82]: right, removed

> Why is this required when. you are providing `version('auto')`

right, removed

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 4 above)
Comment (freschri) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:258]: done

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 3 above)
Comment (freschri) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:20]: done

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 2 above)
Comment (freschri) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:18]: done

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 1 above)
Comment (freschri) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:7]: done

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md [hunk 6]
```diff
@@ -0,0 +1,260 @@
+# Single New EKS Cluster Open Source Observability Accelerator for NVIDIA GPU-based clusters
+
+Graphics Processing Units (GPUs) play an integral part in the Machine Learning (ML) workflow, by providing the scalable performance needed for fast ML training and cost-effective ML inference. On top of that, they are used in flexible remote virtual workstations and powerful HPC computations.
+
+This pattern shows you how to monitor the performance of the GPUs units, used in an Amazon EKS cluster leveraging GPU-based instances.
+
+Amazon Managed Service for Prometheus (AMP) and Amazon Managed Grafana (AMG) are open source tools used in this pattern to collect and visualise metrics respectively.
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:7]: AMP and AMG are not official names of those managed OSS products, i used it AMP and AMG just for short form for PR. Lets remove AMP and AMG references and call it with full names.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
Minor comment.

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-gpu-opensource-observability-accelerator destroy

**Comment by elamaran11** [comment]
/do-e2e-test single-new-eks-gpu-opensource-observability destroy

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md [hunk 7]
```diff
@@ -0,0 +1,260 @@
...
+![GPU_dcgm_Dashboard_3](../images/gpu/gpu_dcgm_3.png)
+
+Grafana Operator and Flux always work together to synchronize your dashboards with Git. If you delete your dashboards by accident, they will be re-provisioned automatically.
+    
+## Teardown
+
+You can teardown the whole CDK stack with the following command:
+
+```bash
+make pattern single-new-eks-gpu-opensource-observability destroy
```
Comment (elamaran11) [inline docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:259]: this should be `make pattern single-new-eks-gpu-opensource destroy`

**Review by elamaran11 — CHANGES_REQUESTED** [review]
Another one

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 7 above)
Comment (freschri) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:259]: this is the output of make list:
Patterns: 

	existing-eks-awsnative-observability
	existing-eks-mixed-observability
	existing-eks-opensource-observability
	single-new-eks-awsnative-observability
	single-new-eks-cluster
	single-new-eks-gpu-opensource-observability
	single-new-eks-graviton-opensource-observability
	single-new-eks-mixed-observability
	single-new-eks-opensource-observability

so for consistency the pattern is called ..-observability, so "make pattern single-new-eks-gpu-opensource-observability destroy" is correct?

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 6 above)
Comment (freschri) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:7]: they actually are :) ?
see here:
https://aws.amazon.com/about-aws/whats-new/2021/05/aws-announces-a-price-reduction-for-amazon-managed-service-for-prometheus-amp/
https://aws.amazon.com/grafana/sla/

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 7 above)
Comment (elamaran11) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md:259]: Well, i got confused, nm. looks good

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 6 above)
Comment (elamaran11) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md]: These docs needs to be updated, Service team does not want to reference these services as AMP or AMG. I have got shot multiple times in blogs. So lets change it

**Review by elamaran11 — CHANGES_REQUESTED** [review]
Minor comments.

File: docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md (hunk 6 above)
Comment (freschri) [reply docs/patterns/single-new-eks-observability-accelerators/single-new-eks-gpu-opensource-observability.md]: done

**Review by elamaran11 — APPROVED** [review]
LGTM
